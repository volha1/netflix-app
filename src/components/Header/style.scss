@import '../../common/scss/base.scss';

.header {
  height: 360px;
  background-image: url('../../common/assets/images/header_bg.png');
  background-color: rgba(0, 0, 0, 0.9);
  background-repeat: no-repeat;
  background-size: cover;
  width: 100%;
  padding: 20px 0px;
}
