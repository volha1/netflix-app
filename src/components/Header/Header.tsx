import React, { ReactElement } from 'react';
import SearchBar from '../SearchBar';
import TopHeader from '../TopHeader';
import './style.scss';

type HeaderProps = {
  handleAddMovieForm: () => void;
  isVisible: boolean;
};

const Header = ({ handleAddMovieForm, isVisible }: HeaderProps): ReactElement | null => {
  return isVisible ? (
    <div className="header">
      <div className="content">
        <TopHeader handleMovieForm={handleAddMovieForm} />
        <SearchBar />
      </div>
    </div>
  ) : null;
};

export default Header;
