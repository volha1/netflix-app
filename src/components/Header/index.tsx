import React from 'react';
import SearchBar from '../SearchBar';
import TopHeader from '../TopHeader';
import './style.scss';

const Header = () => {
  return (
    <div className="header">
      <div className="content">
        <TopHeader />
        <SearchBar />
      </div>
    </div>
  );
};

export default Header;
