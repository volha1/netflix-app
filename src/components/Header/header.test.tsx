import React from 'react';
import renderer from 'react-test-renderer';

import Header from './Header';

describe('Header test', () => {
  const onAddMovieForm = jest.fn();
  const setParams = jest.fn();
  const removeSearchParams = jest.fn();
  const params = { filter: '', sortOrder: '', sortBy: '', search: '', movie: '' };

  it('CloseBtn snapshot', () => {
    const component = renderer.create(
      <Header
        isVisible
        onAddMovieForm={onAddMovieForm}
        params={params}
        setParams={setParams}
        removeSearchParams={removeSearchParams}
      />
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
