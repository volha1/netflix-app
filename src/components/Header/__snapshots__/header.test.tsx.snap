// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test CloseBtn snapshot 1`] = `
<div
  className="header"
>
  <div
    className="content"
  >
    <div
      className="top-header"
    >
      <p
        className="logo"
      >
        <span>
          netflix
        </span>
        roulette
      </p>
      <button
        className="btn add-btn"
        onClick={[MockFunction]}
        type="button"
      >
        + Add movie
      </button>
    </div>
    <div
      className="search-bar"
    >
      <h1
        className="title"
      >
        Find your movie
      </h1>
      <input
        className="search-field"
        onChange={[Function]}
        placeholder="What do you want to watch?"
        value=""
      />
      <button
        className="search-btn"
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
  </div>
</div>
`;
